src/plugins/usage_collection/README.mdx

```diff
@@ -112,6 +113,104 @@ Not an API as such. However, Data Telemetry collects the usage of known patterns
 
 This collector does not report the name of the indices nor any content. It only provides stats about usage of known shippers/ingest tools.
 
+#### Usage Counters
+
+Usage counters allows plugins to report user triggered events from the server. This api has feature parity with UI Counters on the `public` plugin side of usage_collection.
+
+Usage counters provide instrumentation on the server to count triggered events such as "api called", "threshold reached", and miscellaneous events count.
+
+It is useful for gathering _semi-aggregated_ events with a per day granularity.
+This allows tracking trends in usage and provides enough granularity for this type of telemetry to provide insights such as
+- "How many times this threshold has been reached"
+- "What is the trend in usage of this api"
+- "How frequent are users hitting this error per day"
+- "What is the success rate of this operation"
+- "Which option is being selected the most/least"
+
+##### How to use it
+
+To create a usage counter for your plugin, use the API `usageCollection.createUsageCounter` as follows:
+
+```ts
+// server/plugin.ts
+import { Plugin, CoreStart } from '../../../core/public';
+import { UsageCollectionSetup, UsageCounter } from '../../../plugins/usage_collection/server';
+
+export class MyPlugin implements Plugin {
+  private usageCounter?: UsageCounter;
+  public setup(
+    core: CoreStart,
+    { usageCollection }: { usageCollection?: UsageCollectionSetup }
+  ) {
+
+    /**
+     * Create a usage counter for this plugin. Domain ID must be unique.
+     * It is advised to use the plugin name as the domain ID for most cases.
```

Comment:
We allow `:` in `domainId` but not in `counterType`. This allows ui_counters to not have this limitation at all since `counterType` in ui counters are predefined metric types (loaded, count, etc).

Having this limitation does not bother me honestly. we can prevent it by using a more obsecure character like `|_|` or something.

I think we can address it for both ui_counters and usage_counters at a later stage if it becomes bothersome. what do you think?